### ISSUE TITLE
API connection errors (connection refused/connection rest by peer) - gloo-ee 1.12.10

### ISSUE THREAD

**Issue by yarinisraeli24** [solo-io/gloo#7143]
### Gloo Edge Version

1.12.x (latest stable)

### Kubernetes Version

1.23.x

### Describe the bug

Hi colleagues,

I am trying to upgrade the gloo-ee and gloo-fed version from 1.11.37 to 1.12.10 and I got connection errors for my API resources (connection refused/connection reset by peer).

This is the log from the gateway-proxy pod:

```
[2022-09-14 08:02:00.506][6][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:126] gRPC config for [type.googleapis.com/envoy.config.cluster.v3.Cluster](http://type.googleapis.com/envoy.config.cluster.v3.Cluster) rejected: Error adding/updating cluster(s) health-dev-s3_edge-domains: Didn't find a registered implementation for 'envoy.http.stateful_header_formatters.preserve_case' with type URL: 'envoy.extensions.http.header_formatters.preserve_case.v3.PreserveCaseFormatterConfig', e2e-test-api-us_t3c-e2e-consumer: Didn't find a registered implementation for 'envoy.http.stateful_header_formatters.preserve_case' with type URL: 'envoy.extensions.http.header_formatters.preserve_case.v3.PreserveCaseFormatterConfig'
```

hybridGateway configured in gateway-proxy-ssl as you can see here.

```yaml
hybridGateway:
    delegatedHttpGateways:
      httpConnectionManagerSettings:
        enableTrailers: true
        idleTimeout: 300s
        preserveCaseHeaderKeyFormat: true
        serverHeaderTransformation: PASS_THROUGH
        useRemoteAddress: true
```

For more details/resources you can contact me on SAP slack channel and I will provide you.

### Steps to reproduce the bug

N/A

### Expected Behavior

N/A

### Additional Context

N/A

**Comment by nfuden** [comment]
TLDR: Fixed in gloo-ee 1.12.16
We rely on go-control-plane for some of our operations and it links to certain  envoyproxy protos. They now rely on automatic updates from envoy protos but did not previously. This meant that our base version of envoyproxy and the protos we were using from control plane were slightly out of line. This should not be a problem anymore and we will write a test to prevent regressions
